### ISSUE TITLE
Form collection validating first item only

### ISSUE THREAD

**Issue by kzweb** [kristijanhusak/laravel-form-builder#268]
![collection-validation](https://cloud.githubusercontent.com/assets/10473169/17647602/9e0af478-61cd-11e6-9f08-839435512b2e.jpg)

```php
class CarroTagForm extends Form
{
    public function buildForm()
    {
        $this
            ->add('ntag', 'text', [
                'label' => 'Nome da Tag',
                'rules' => 'required|min:5',
                'error_messages' => [
                    'ntag.required' => 'O nome da tag é obrigatório.'
                ]
            ])
            ;
    }
}
````

```php
class CarroForm extends Form
{
    public function buildForm()
    {
        $this
            ->add('nome', 'text', [
                'label' => 'Nome do Carro',
                'rules' => 'required|min:5',
                'error_messages' => [
                    'title.required' => 'O nome do carro é obrigatório.'
                ]
            ])
            ->add('tags', 'collection', [
                'type' => 'form',
                'options' => [
                    'class' => CarroTagForm::class,
                    'label' => false,
                ]
            ])
            ->add('submit', 'submit', [
                'label' => 'Salvar'
            ])
            ;
    }
}
```

Manual validation:

```php
{
        $rules = [
            'name' => 'required|min:10',
        ];

        foreach ($request->get('tags') as $key => $fields) {
            $fields = array_keys($fields);
            foreach ($fields as $field) {
                $rules["tags.$key.$field"] = 'required|max:10';
            }
        }

        $validator = Validator::make($request->all(), $rules);

        if ($validator->fails()) {
            return redirect()
                ->back()
                ->withErrors($validator)
                ->withInput();
        }

        dd('validated');
```

Results:

![2](https://cloud.githubusercontent.com/assets/10473169/17650592/969d6afe-6227-11e6-96a1-d80a78f13110.png)

**Comment by kristijanhusak** [comment]
I fixed this on `dev-master`. Collection type only collected data from old request data, which is available only after redirect.
Will be in next patch version after few other things are fixed.
Thanks for reporting.

**Comment by kzweb** [comment]
Thank you. Already tested and is working for me.

**Comment by kristijanhusak** [comment]
This is now available on stable version 1.9.
